tooling/minimize/tests/pass/iter.rs

```diff
@@ -0,0 +1,31 @@
+
+extern crate intrinsics;
+use intrinsics::*;
+
+struct RepeatN {
+    val: bool,
+    repetitions: u8,
+}
+
+impl Iterator for RepeatN {
+    type Item = bool;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.repetitions > 0 {
+            self.repetitions -= 1;
+            Some(self.val)
+        } else {
+            None
+        }
+    }
+}
+
+fn main() {
+    let iter = RepeatN {
+        val: true,
+        repetitions: 3,
+    };
+    for i in iter {
+        print(i as u8);
```

Comment:
Oh, I thought this was meant to print 0, 1, 2?

Reply:
Funnily enough I found a bug when I converted the `bool` to `u8` as the cast is a `IntToInt`-cast internally. So I'd like to keep it this way to have this as a test somewhere. Also the iterator is called `RepeatN`, of course it repeats it's value. We can test that it repeats the right amount of times by checking that the value appears N times in the output.

Reply:
> Funnily enough I found a bug when I converted the bool to u8 as the cast is a IntToInt-cast internally. So I'd like to keep it this way to have this as a test somewhere. 

If you want a bool2int cast test, that should be a dedicated test (in tooling/minimize/tests/pass/casts.rs), not an accidental side-effect of another test. Otherwise the chances are too high that this will be refactored away at some point.

Reply:
Repeating a bool here seems a bit odd as well, why not repeat a `u8`?

> We can test that it repeats the right amount of times by checking that the value appears N times in the output.

Relying on output for testing is not great. When a test fails it's not easy to figure out where things even went wrong. You have to figure out which part of the output corresponds to which lines of code.

It's much better to have the assertions directly in the test.